compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1334,12 +1334,9 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
 
             val setter = toSetter(lhsCore)
             if setter.isEmpty then reassignmentToVal
```

Comment:
When can `setter` actually be empty? We haven't found any such cases.

Reply:
When the pattern match in `toSetter` gets to the default case. But I am not sure this is ever the case. On the other hand, programs with parse errors can give uncommon trees, so better be defensive.